### PR TITLE
Added an output pydy-optional for installing all optional dependencies.

### PR DESCRIPTION
**Pull request by moorepants**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by chrisburr** [comment]
Following on from the discussion on gitter I've made a PR to this PR to show what I meant more clearly. See https://github.com/moorepants/pydy-feedstock/pull/1.

**Comment by moorepants** [comment]
@chrisburr Thanks. I commented on your PR and also updated this to do what I think I want it to do.

**Comment by moorepants** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,10 +48,46 @@ test:
...
+    version: {{ version }}
+    requirements:
+      run:
+        - python
+        - pydy {{ version }}
+        # Optional dependencies:
+        - notebook 4.*
+        - ipywidgets 4.*
+        - cython >=0.23.4
+        - theano >=0.8.0
```
Comment (moorepants) [inline recipe/meta.yaml:60]: These would be enforced by the run_constrained above, so I could leave off the specifiers.

**Comment by chrisburr** [comment]
I think you need both `python` and `pip` in the `host` section of the `requirements`.

**Comment by moorepants** [comment]
Ok, I think I'm happy with this. Merging.

**Pull request merged**